dafny-lang/dafny
# Various fixes for project aware mode

### Changes
1. Correctly handle publication of diagnostics that don't belong to the current project, using `the referenced file {uri.LocalPath} contains error(s) but is not owned by this project. The first error is: {errors.First().Message}`
1. Take URIs into account when handling manual verification run and cancellation requests
1. Enable project aware mode option
1. Send compilation status updates for all documents owned by a project
1. Do not retrigger compilation of a project when opening an unchanged document of that project

### Tests
1. Removed the `Skip` attribute from two existing tests
1. Add test `RunWithMultipleSimilarDocuments`
1. No test needed
1. Add test `CompilationStatusNotificationTest.MultipleDocuments`
1. Performance, difficult to test without an existing framework

<small>By submitting this pull request, I confirm that my contribution is made under the terms of the [MIT license](https://github.com/dafny-lang/dafny/blob/master/LICENSE.txt).</small>
